packages/a11y-base/src/list-mixin.js

```diff
@@ -117,8 +118,15 @@ export const ListMixin = (superClass) =>
       if (this._observer) {
         this._observer.flush();
       }
-      const firstItem = this.querySelector('[tabindex="0"]') || (this.items ? this.items[0] : null);
-      this._focusItem(firstItem);
+
+      const items = Array.isArray(this.items) ? this.items : [];
+      const idx = this._getAvailableIndex(items, 0, null, (item) => item.tabIndex === 0 && !isElementHidden(item));
```

Comment:
By the way, I've just noticed that `_getAvailableIndex` already takes into account the disabled state while going through the items, which means the else branch is probably useless.